jetstream_pt/engine.py

```diff
@@ -600,23 +602,41 @@ def colocated_cpus(self) -> Union[list[engine_api.CpuDevices], None]:
 
   def get_prefix_destination_sharding(self) -> Prefix:
     """Returns the shardings necessary to transfer data between engines."""
-    return Prefix(
-        self.replicated,
-        self.cache_sharding,
-        self.replicated,
-    )
+    if self.env.shard_on_batch:
+      return Prefix(
+          self.replicated,  # cache is replicated because bs=1 for prefill
```

Comment:
The multiple prefill isolated instance's performance is better than this replicated sharding (or we use samller vm to test instead of 4chips or 8 chips).

Reply:
Hi Fanhai, do you mean each TPU chip do certain length of the prefill?

Reply:
do you mean having 8 InterleavedEngine with one device each; vs. sharding on batch from jax?

Reply:
The 8 InterleavedEngine vs sharding on batch from jax.

Reply:
Of course because there is no collective operations involved. IIUC then the best config might be enable disaggregated serving and use multiple prefill engine instances instead of sharding.